scripts/zones/Lower_Jeuno/npcs/Darcia.lua

```diff
@@ -0,0 +1,59 @@
+-----------------------------------
+-- Area: Lower Jeuno (245)
+--  NPC: Darcia
+-- SoA Mission NPC
+-- pos  25 -38.617 -1.000
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/missions")
+require("scripts/globals/npc_util")
+require("scripts/globals/settings")
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local rumorsFromTheWest = player:getCurrentMission(SOA) == dsp.mission.id.soa.RUMORS_FROM_THE_WEST
+    local theGeomagnetron = player:getCurrentMission(SOA) == dsp.mission.id.soa.THE_GEOMAGNETRON
+
+    if not ENABLE_SOA then
+        player:startEvent(10124)
+    elseif rumorsFromTheWest then
+        player:startEvent(10117, 0, 40, 26, 0)
+    elseif theGeomagnetron and player:getVar("SOA") == 1 then
+        player:startEvent(10118)
+    elseif theGeomagnetron then
+        player:startEvent(10117, 1, 40, 26, 0)
+    else
+        player:startEvent(10123)
+    end
+end
+
+function onEventUpdate(player,csid,option)
+end
+
+function onEventFinish(player,csid,option)
+    if csid == 10117 and option == 1 then -- accepted geomagnetron
+        npcUtil.giveKeyItem(player, dsp.ki.GEOMAGNETRON)
+
+        player:completeMission(SOA,dsp.mission.id.soa.RUMORS_FROM_THE_WEST)
+        player:addMission(SOA,dsp.mission.id.soa.THE_GEOMAGNETRON)
+    elseif csid == 10117 and option == 2 then -- paid or
```

Comment:
I don't think there'd be harm in setting the player's SOA var to 0 if they paid (since I believe it'd already be zero), so you might be able to combine these branches into one.